src/test/java/edu/ucla/library/prl/harvester/InstitutionTest.java

```diff
@@ -0,0 +1,228 @@
+
+package edu.ucla.library.prl.harvester;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import javax.mail.internet.AddressException;
+import javax.mail.internet.InternetAddress;
+
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import com.google.i18n.phonenumbers.NumberParseException;
+import com.google.i18n.phonenumbers.PhoneNumberUtil;
+import com.google.i18n.phonenumbers.PhoneNumberUtil.PhoneNumberFormat;
+import com.google.i18n.phonenumbers.Phonenumber.PhoneNumber;
+
+import io.vertx.core.json.JsonObject;
+import io.vertx.junit5.VertxExtension;
+
+/**
+ * Tests {@link Institution}.
+ */
+@ExtendWith(VertxExtension.class)
+public class InstitutionTest {
+
+    /**
+     * Parses and formats phone numbers.
+     */
+    private static final PhoneNumberUtil PHONE_NUMBER_UTIL = PhoneNumberUtil.getInstance();
+
+    /**
+     * Tests that an {@link Institution} can be instantiated from a {@link JsonObject} and serialized back to one.
+     *
+     * @param aName The institution's name
+     * @param aDescription The institution's description
+     * @param aLocation The institution's human-readable location
+     * @param anEmail The optional email address contact of the institution
+     * @param aPhone The optional phone number contact of the institution
+     * @param aWebContact The optional web contact of the institution
+     * @param aWebsite The institiution's website
+     */
+    @ParameterizedTest
+    @MethodSource
+    void testInstitutionSerDe(final String aName, final String aDescription, final String aLocation,
```

Comment:
Well, you guessed right ;) (although [I can't take credit for the name](https://serde.rs/))

Although, I can see that the name could be better; it's not just testing serialization/deserialization, but rather it's asserting that the Institution (and Job) types work in the ways we expect. I kind of like having all this in one method since it's rather short at ~ 25 LOC, but perhaps it could use more inline comments?

Reply:
"work in the ways we expect" -- so this is considered a test of the getters/setters too? I definitely like splitting those out in keeping with the "test one thing at a time" approach. My 2c.

Reply:
If you keep it, yes, I guess a comment indicating this is testing setters/getters in addition to (de)serialization.

Reply:
There are no setters since these types are immutable; having individual test methods for getters would make the test class longer for a benefit that is unclear to me.

Reply:
And yes, I would consider this a test of the getters too; I think testing ser/de implies testing getters (i.e., ser/de works if calling a getter on the deserialized-from-JSON instance returns the same value as calling that getter on the constructor-instantiated instance).

Reply:
Additional comments added